crates/ruff_dev/src/generate_ty_options.rs

```diff
@@ -25,6 +25,10 @@ pub(crate) fn main(args: &Args) -> anyhow::Result<()> {
     let file_name = "crates/ty/docs/configuration.md";
     let markdown_path = PathBuf::from(ROOT_DIR).join(file_name);
 
+    output.push_str(
+        "<!-- WARNING: This file is auto-generated. Edit 'generate_ty_options.rs' instead. -->\n\n",
```

Comment:
Same here, users shouldn't have to edit the script. Instead, they should update the documention on the `Options` struct in `ty_project`.